main.nf

```diff
@@ -14,201 +14,183 @@ Sabrina Krakau <sabrinakrakau@gmail.com>
+        .set { ch_input_rows }
+    // separate short and long reads
+    ch_input_rows
+        .map { id, group, sr1, sr2, lr ->
+                    def meta = [:]
+                    meta.id           = id
+                    meta.group        = group
+                    meta.single_end   = params.single_end
+                    if (params.single_end) 
+                        return [ meta, [ sr1] ]
+                    else 
+                        return [ meta, [ sr1, sr2 ] ]
+            }
         .set { ch_raw_short_reads }
-    ch_all_files_lr
-        .map { row -> if (row.size() == 5) [ row[0], row[1], row[4] ] }
+    ch_input_rows
+        .map { id, group, sr1, sr2, lr ->
+                    if (lr) {
+                        def meta = [:]
+                        meta.id           = id
+                        meta.group        = group
+                        return [ meta, lr ]
+                    }
+            }
         .set { ch_raw_long_reads }
-} else if(params.input_paths){
-    if(params.single_end){
-        Channel
-            .from(params.input_paths)
-            .map { row -> [ row[0], 0, [ file(row[1][0], checkIfExists: true) ] ] }
-            .ifEmpty { exit 1, "params.input_paths was empty - no input files supplied" }
-            .into { ch_sample_validate; ch_raw_short_reads }
-        ch_raw_long_reads = Channel.empty()
-    } else {
-        Channel
-            .from(params.input_paths)
-            .map { row -> [ row[0], 0, [ file(row[1][0], checkIfExists: true), file(row[1][1], checkIfExists: true) ] ] }
-            .ifEmpty { exit 1, "params.input_paths was empty - no input files supplied" }
-            .into { ch_sample_validate; ch_raw_short_reads }
-        ch_raw_long_reads = Channel.empty()
-    }
- } else {
+} else {
     Channel
         .fromFilePairs(params.input, size: params.single_end ? 1 : 2)
         .ifEmpty { exit 1, "Cannot find any reads matching: ${params.input}\nNB: Path needs to be enclosed in quotes!\nIf this is single-end data, please specify --single_end on the command line." }
-        .map { row -> [ row[0], 0, row[1] ] }
+        .map { row -> 
+                    def meta = [:]
+                    meta.id           = row[0]
+                    meta.group        = 0
+                    meta.single_end   = params.single_end
+                    return [ meta, row[1] ]
+            }
         .set { ch_raw_short_reads }
-    ch_sample_validate = Channel.empty()
+    ch_input_rows = Channel.empty()
     ch_raw_long_reads = Channel.empty()
 }
 
 // Ensure sample IDs are unique
-ch_sample_validate
-    .map { row -> row[0] }
+ch_input_rows
+    .map { id, group, sr1, sr2, lr -> id }
     .toList()
-    .map{ ids -> if( ids.size() != ids.unique().size() ) {exit 1, "ERROR: input contains duplicated sample IDs!" } }
+    .map { ids -> if( ids.size() != ids.unique().size() ) {exit 1, "ERROR: input samplesheet contains duplicated sample IDs!" } }
```

Comment:
This could go into a separate file.

Reply:
Hi @d4straub and @KevinMenden ,
were should this go, into a separate module file or should this be added to `functions.nf`?

Reply:
I would put it into a separate file.

Reply:
should go into a proper `NF` script in `modules/local`